typescript/examples/validate/scripts/validate.ts

```diff
@@ -139,7 +139,7 @@ async function validate(
           throw new Error(`Missing schema ${record.schemaId} for channel ${record.id}`);
         }
         let messageDeserializer: (data: ArrayBufferView) => unknown;
-        if (record.messageEncoding === "ros1") {
+        if (record.messageEncoding === "ros1" && schema.encoding === "ros1msg") {
```

Comment:
nitty nit: feels a bit more natural to me to check schema first, then the message encoding, going from less specific to more specific.

Reply:
https://github.com/foxglove/mcap/assets/14237/968f69bf-3fb4-4695-b796-6456c5a73472